### PR TITLE
python.pkgs.protobuf: Fix cpp backend

### PR DESCRIPTION
**Pull request by knedlsepp**
###### Motivation for this change
This fixes the python-protobuf C++ backend which is way faster than the python backend.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by knedlsepp** [comment]
@ofborg build pkgsCross.aarch64-multiplatform.python3Packages.protobuf

(To check if the introduction of the dev-output in #99141 was really doing something for cross)

**Comment by veprbl** [comment]
@GrahamcOfBorg eval

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.protobuf python39Packages.protobuf

File: pkgs/development/python-modules/protobuf/default.nix [hunk 1]
```diff
@@ -42,27 +40,12 @@ buildPythonPackage {
...
-  # install, please do replace this.
-  postInstall = lib.optionalString (lib.versionAtLeast protobuf.version "2.6.0") ''
-    cp -v $(find build -name "_message*") $out/${python.sitePackages}/google/protobuf/pyext
-  '';
+  pythonImportsCheck = lib.optionals (lib.versionAtLeast protobuf.version "2.6.0") [
+    "google.protobuf.internal._api_implementation" # Verify that --cpp_implementation worked
+  ];
 
   meta = with lib; {
     description = "Protocol Buffers are Google's data interchange format";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protobuf/default.nix:51]: The package has no maintainer. Do you want to add yourself?

File: pkgs/development/python-modules/protobuf/default.nix [hunk 2]
```diff
@@ -42,27 +40,12 @@ buildPythonPackage {
...
+    "--cpp_implementation";
 
-  # the _message.so isn't installed, so we'll do that manually.
-  # if someone can figure out a less hacky way to get the _message.so to
-  # install, please do replace this.
-  postInstall = lib.optionalString (lib.versionAtLeast protobuf.version "2.6.0") ''
-    cp -v $(find build -name "_message*") $out/${python.sitePackages}/google/protobuf/pyext
-  '';
+  pythonImportsCheck = lib.optionals (lib.versionAtLeast protobuf.version "2.6.0") [
+    "google.protobuf.internal._api_implementation" # Verify that --cpp_implementation worked
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protobuf/default.nix:47]:
```suggestion
  pythonImportsCheck = [
    "google.protobuf"
  ] ++ lib.optionals (lib.versionAtLeast protobuf.version "2.6.0") [
    "google.protobuf.internal._api_implementation" # Verify that --cpp_implementation worked
```

File: pkgs/development/python-modules/protobuf/default.nix [hunk 3]
```diff
@@ -19,8 +19,6 @@ buildPythonPackage {
   inherit disabled;
   doCheck = doCheck && !isPy27; # setuptools>=41.4 no longer collects correctly on python2
 
-  outputs = [ "out" "dev" ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/protobuf/default.nix:22]: dev is empty. 👍🏼

Comment (knedlsepp) [reply pkgs/development/python-modules/protobuf/default.nix:22]: I also read that currently python packages break when they have a dev output. So even if it weren't empty we probably should have dropped it. Cannot point you to the exact issue, sorry.

**Comment by knedlsepp** [comment]
@SuperSandro2000 Sorry for the delay. I resolved the remaining comments.

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.protobuf python39Packages.protobuf

**Pull request merged**